[snowflake][ccm] Add query tag documentation (#27774)
DataDog/documentation: content/en/cloud_cost_management/saas_costs.md

```
ALTER SESSION SET QUERY_TAG = '{"team": "devops", "application": "CI_job_executor"}';
```
{{< img src="cloud_cost/saas_costs/snowflake_query_tags_example.png" alt="Group costs by team and application query tags." style="width:100%" >}}

As a result, the costs of all queries executed with the `team` and `application` query tags will be attributable to those concepts.

In order to leverage query tags within cost management, you must ensure that the following rules are adhered to.

First, the query_tag string must be JSON parsable. Specifically, this means that the string is processable by the native PARSE_JSON function.

Secondly, an allowlist of keys must be provided in the Snowflake integration tile. These keys map to the first layer of the JSON formatted query_tag field. This allowlist will appear in the form of a comma-separated list of strings, e.g. tag_1,tag_2,tag_3. Please ensure that strings are only alphanumeric characters, or underscores, hyphens, and periods. This information can be entered into the Snowflake tile, under “Resources Collected -> Cloud Cost Management -> Collected Query Tags”

> Notes:
> - **Optimizing tags**: Although a powerful concept, query tags should be selected with data magnitude in mind. Appropriate query tags are ones that have low to medium group cardinality (e.g. team, user, service). As the number of unique values increases, bottlenecking issues for both data ingestion and frontend rendering can manifest. An example of a suboptimal query tag is the unique UUID associated with job executions.

Reviewer: ```suggestion
**Note**: Select your query tags with data magnitude in mind. Appropriate query tags are ones that have low to medium group cardinality (for example: team, user, service). Selecting a query tag with high group cardinality (such as unique UUID associated with job executions) can result in bottlenecking issues for both data ingestion and frontend rendering.
```
